### PR TITLE
Command

### PR DESCRIPTION
**Pull request by shinsuke-nara**
I designed new API to post new command with title, description and meta data.

Could you review this?

This PR must not be merged.

Related issue No 788.

### CODE REVIEW

**Comment by shinsuke-nara** [comment]
@satoshikumano Could you review this?

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 1]
```diff
@@ -416,9 +417,11 @@ public void uninstallPush(@NonNull String installationID) throws ThingIFExceptio
      * of the command by calling {@link #getCommand}.
      * @throws ThingIFException Thrown when failed to connect IoT Cloud Server.
      * @throws ThingIFRestException Thrown when server returns error response.
+     * @deprecated use {@link ThingIFAPI#postNewCommand(CommandForm)}.
```
Comment (satoshikumano) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: Lets' not deprecated this method.
Still useful when Title/ Metadata/ Description is not required.

**Comment by shinsuke-nara** [comment]
@satoshikumano https://github.com/KiiPlatform/thing-if-AndroidSDK/pull/75#discussion_r58328207
I will remove deprecated mark.

**Comment by shinsuke-nara** [comment]
@satoshikumano I removed deprecated from `postNewCommand()`.

I also added getter of fields. They are required to get values in CommandFrom at ThingIFAPI class.
I think such getters is not required as public API, but to get these values, They are required to be public API.

If we can move CommandForm class to `com.kii.thingif` package, These getters can be package internal. I think it is better idea.

What do you think?

**Comment by shinsuke-nara** [comment]
@naoki-akimoto Could you implement this feature? Please create new brunch and make PR to this brunch.

**Comment by satoshikumano** [comment]
@shinsuke-nara LGTM

File: thingif/src/main/java/com/kii/thingif/command/CommandForm.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        // TODO: validate following fields if invalid, throw exception.
+        this.schemaName = schemaName;
+        this.schemaVersion = schemaVersion;
+        this.actions = actions;
+    }
+
+    /**
+     * Setter of title
+     *
+     * @param title Length of title must be equal or less than 50.
```
Comment (shinsuke-nara) [inline thingif/src/main/java/com/kii/thingif/command/CommandForm.java]: 50 characters.

**Comment by shinsuke-nara** [comment]
Spec review is finished.

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 3]
```diff
@@ -447,6 +448,57 @@ public Command postNewCommand(
...
+            throw new UnsupportedSchemaException(schemaName, schemaVersion);
+        }
+
+        String path = MessageFormat.format("/thing-if/apps/{0}/targets/{1}/commands",
+                this.app.getAppID(), this.target.getTypedID().toString());
+        String url = Path.combine(this.app.getBaseUrl(), path);
+        Map<String, String> headers = this.newHeader();
+        Command command = new Command(schemaName, schemaVersion, this.owner.getTypedID(),
+                form.getActions());
+        if (form.getTitle() != null) {
```
Comment (nfukuzaki) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: Nullableなのでif文は不要な気がします。
description, metadataも同様です。

Comment (shinsuke-nara) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: @nfukuzaki  commandのsetterがnullを受け付けられるので、if文は不要ですね。

@naoki-akimoto 対応お願いします。

**Comment by naoki-akimoto** [comment]
@shinsuke-nara @nfukuzaki 
不要なif文の削除しました。

**Comment by nfukuzaki** [comment]
LGTM

**Pull request merged**